docs/source/conf.py

```diff
@@ -85,6 +85,23 @@ def __getattr__(self, name):
     def __setattr__(self, name, value):
         return setattr(sys.stdout, name, value)
 pwnlib.context.ContextType.defaults['log_console'] = stdout()
+
+offline=True
+binutils_mips=False
+binutils_arm=False
+binutils_amd64=False
+binutils_i386=False
+binutils_aarch64=False
+binutils_powerpc=False
+binutils_thumb=False
+qemu_mips=False
+qemu_arm=False
+qemu_amd64=False
```

Comment:
All instances of {qemu,binutils}_{i386,amd64} should be removed entirely.  These are always available.  If they are not available, you're running Pwntools from an unsupported machine type and the tests are not expected to work.

Reply:
They are not always available. The intention is to be able to verify that basic pwntools functionality works as intended, even if for example no mips binutils are available.

Are multiple people using this account? I feel like your comments contradict your previous sentiment.

Reply:
Please re-read my comment.  It does not refer to MIPS binutils, but specifically the i386 and amd64 binutils and QEMU.

No valid testing environment exists where you cannot execute i386 and amd64 binaries, as a **requirement** for using Pwntools is to use amd64 as the host architecture.  Other architectures *may* work, but it is not worth the support burden to the maintenance team to ensure they work.

Separately, in a testing environment, **why** is the MIPS toolchain not available?  We have the ability to install any packages we want.

This is my personal account.  I am currently the only Pwntools maintainer.